python/foglamp/services/core/user_model.py

```diff
@@ -347,6 +350,49 @@ class Objects:
 
             return uid, jwt_token, False
 
+        @classmethod
+        async def certificate_login(cls, username, host):
+            """
+            Args:
+                user_cert: user_cert
+            Returns:
+                  return token
+
+            """
+            storage_client = connect.get_storage_async()
+
+            # get user info on the basis of username
+            payload = PayloadBuilder().SELECT("id", "role_id").WHERE(['uname', '=', username])\
+                .AND_WHERE(['enabled', '=', 't']).payload()
+            result = await storage_client.query_tbl_with_payload('users', payload)
+            if len(result['rows']) == 0:
+                raise User.DoesNotExist('User does not exist')
+
+            found_user = result['rows'][0]
+
+            # fetch user info
+            exp = datetime.now() + timedelta(seconds=JWT_EXP_DELTA_SECONDS)
+            uid = found_user['id']
+            p = {'uid': uid, 'exp': exp}
+            jwt_token = jwt.encode(p, JWT_SECRET, JWT_ALGORITHM).decode("utf-8")
+
+            payload = PayloadBuilder().INSERT(user_id=p['uid'], token=jwt_token,
+                                              token_expiration=str(exp), ip=host).payload()
+
+            # Insert token, uid, expiration into user_login table
+            try:
+                await storage_client.insert_into_tbl("user_logins", payload)
+            except StorageServerError as ex:
+                if ex.error["retryable"]:
+                    pass  # retry INSERT
+                raise ValueError(ERROR_MSG)
+
+            # TODO remove hard code role id to return is_admin info
+            if int(found_user['role_id']) == 1:
+                return uid, jwt_token, True
+
+            return uid, jwt_token, False
```

Comment:
looks like this is 80% duplicate of login def. Please put common code in _login def?